app/assets/javascripts/app/views/unblock_person_view.js

```diff
@@ -0,0 +1,24 @@
+// @license magnet:?xt=urn:btih:0b31508aeb0634b347b8270c7bee4d411b5d4109&dn=agpl-3.0.txt AGPL-v3-or-Later
+
+app.views.UnblockPerson = app.views.Base.extend({
+  templateName: "unblock_person",
+
+  events: {
+    "click #unblock_user_button": "unblock"
+  },
+
+  initialize: function(opts) {
+    _.extend(this, opts);
+    this.parent = opts.parent;
+  },
+
+  unblock: function() {
+    var block = this.person.unblock();
+    block.fail(function() {
+      app.flashMessages.error(Diaspora.I18.t("unblock_failed"));
+    });
+    this.parent._populateHovercard();
```

Comment:
There is a race-condition with this. The unblock happens in parallel with the reload, so when the reload reloads the hovercard.json before the unblock is persisted, the hovercard is stuck in an inconsistent state (happens in about 50% of my tries):

![image](https://user-images.githubusercontent.com/458548/38341534-5cba4f44-3879-11e8-89ad-de360eb41d7b.png)

There is an event triggered when the unblock is done, see:

https://github.com/diaspora/diaspora/blob/b4e3f84607279438096b462b83749da9120bed03/app/assets/javascripts/app/pages/profile.js#L39

Maybe you can do it similar?